### PR TITLE
Run a selected text containing FQL query

### PR DESCRIPTION
**Pull request by jfloresremar**
I think it will be useful when there are more than one query inside a .fql file.
It will run the entire file if there is no selected text in activeEditor.
Thank you for this extension!.

### CODE REVIEW

**Comment by BrunoQuaresma** [comment]
Thanks for the contribution @jfloresremar. It is a nice suggestion!

I'll review it and make it aligned with our code styles before merge.

**Comment by jfloresremar** [comment]
That is good. Thank you!

**Comment by fauna-ryanharris** [comment]
@jfloresremar Thanks for the contribution! Could you please allow us to make changes by following the instructions below?

https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by jfloresremar** [comment]
Hello @fauna-ryanharris
I followed the instructions and i see that Allow edits from maintainers is
checked.
This is first time i've created a fork so if something is missing
please let me know

File: src/runQueryCommand.ts [hunk 1]
```diff
@@ -24,7 +24,21 @@ export default (
...
-  const code = activeTextEditor.document.getText();
+
+  const selection = activeTextEditor.selection;
+  const selectedText = activeTextEditor.document.getText(selection);
+  let code = '';
+  if (selectedText.length > 0) {
+    code = selectedText;
+  } else {
+    code = activeTextEditor.document.getText();
+  }
```
Comment (fauna-ryanharris) [inline src/runQueryCommand.ts]:
```suggestion
  const fql = selectedText.length > 0 ? selectedText : activeTextEditor.document.getText();
```

Comment (fauna-ryanharris) [reply src/runQueryCommand.ts]: We can use a ternary here to handle both the variable assignment and the conditional logic. Let's also update the variable name to be more clear.

File: src/runQueryCommand.ts [hunk 2]
```diff
@@ -24,7 +24,21 @@ export default (
...
+  if (code.length === 0) {
+    vscode.window.showWarningMessage(
+      'Selected file or selected text must have a FQL query to run'
+    );
+    return;
+  }
```
Comment (fauna-ryanharris) [inline src/runQueryCommand.ts]:
```suggestion
  if (fqlExpression.length < 1) {
     vscode.window.showWarningMessage(
      'Selected file or selected text must have a FQL query to run'
    );
    
    return;
  }
```

Comment (fauna-ryanharris) [reply src/runQueryCommand.ts]: Let's update the variable name here too.

File: src/runQueryCommand.ts [hunk 3]
```diff
@@ -24,7 +24,17 @@ export default (
...
+  if (fqlExpression.length < 1) {
+     vscode.window.showWarningMessage(
+      'Selected file or selected text must have a FQL query to run'
+    );
+    
+    return;
+  }
 
   outputChannel.appendLine('');
   outputChannel.appendLine(`RUNNING: ${code}`);
```
Comment (fauna-ryanharris) [inline src/runQueryCommand.ts]:
```suggestion
  outputChannel.appendLine(`RUNNING: ${fqlExpression}`);
```

Comment (fauna-ryanharris) [reply src/runQueryCommand.ts]: We need to update the variable here to match the changes above

**Comment by fauna-ryanharris** [comment]
@danieltodonnell @BrunoQuaresma @fauna-rodrigo I tested this locally by building a new `.vsix` with these changes and executing some queries. Works as expected on my end. Once one of you confirm that, I'll merge 🙏

**Comment by BrunoQuaresma** [comment]
@fauna-ryanharris can you add some notes about this on the README? Would be nice to let us users know about this feature.

**Comment by fauna-ryanharris** [comment]
@BrunoQuaresma Sure. I'll add some notes before I merge the PR 👍 Good call out.

**Pull request merged**